libtock/ieee802154.c

```diff
@@ -67,196 +67,138 @@ bool ieee802154_is_up(void) {
+  if (!addr_long) return RETURNCODE_EINVAL;
+
   allow_rw_return_t rw = allow_readwrite(RADIO_DRIVER, ALLOW_CFG, (void *) addr_long, 8);
-  if (!rw.success) {
-    return tock_error_to_rcode(rw.error);
-  }
+  if (!rw.success) return tock_status_to_returncode(rw.status);
+
   syscall_return_t com = command(RADIO_DRIVER, COMMAND_SET_ADDR_LONG, 0, 0);
-  if (com.type == TOCK_SYSCALL_SUCCESS) {
-    return TOCK_SUCCESS;
-  } else if (com.type > TOCK_SYSCALL_SUCCESS) {
-    return TOCK_FAIL;
-  } else {
-    return tock_error_to_rcode(com.data[0]);
-  }
+  return tock_command_return_novalue_to_returncode(com);
 }
 
 int ieee802154_set_pan(unsigned short pan) {
   syscall_return_t com = command(RADIO_DRIVER, COMMAND_SET_PAN, (unsigned int) pan, 0);
-  if (com.type == TOCK_SYSCALL_SUCCESS) {
-    return TOCK_SUCCESS;
-  } else if (com.type > TOCK_SYSCALL_SUCCESS) {
-    return TOCK_FAIL;
-  } else {
-    return tock_error_to_rcode(com.data[0]);
-  }
+  return tock_command_return_novalue_to_returncode(com);
 }
 
 int ieee802154_set_channel(unsigned char channel) {
   syscall_return_t com = command(RADIO_DRIVER, COMMAND_SET_CHANNEL, (unsigned int) channel, 0);
-  if (com.type == TOCK_SYSCALL_SUCCESS) {
-    return TOCK_SUCCESS;
-  } else if (com.type > TOCK_SYSCALL_SUCCESS) {
-    return TOCK_FAIL;
-  } else {
-    return tock_error_to_rcode(com.data[0]);
-  }
+  return tock_command_return_novalue_to_returncode(com);
 }
 
 int ieee802154_set_power(char power) {
   // Cast the signed char to an unsigned char before zero-padding it.
   syscall_return_t com = command(RADIO_DRIVER, COMMAND_SET_POWER, (unsigned int) (unsigned char) power, 0);
-  if (com.type == TOCK_SYSCALL_SUCCESS) {
-    return TOCK_SUCCESS;
-  } else if (com.type > TOCK_SYSCALL_SUCCESS) {
-    return TOCK_FAIL;
-  } else {
-    return tock_error_to_rcode(com.data[0]);
-  }
+  return tock_command_return_novalue_to_returncode(com);
 }
 
 int ieee802154_config_commit(void) {
   syscall_return_t com = command(RADIO_DRIVER, COMMAND_CONFIG_COMMIT, 0, 0);
-  if (com.type == TOCK_SYSCALL_SUCCESS) {
-    return TOCK_SUCCESS;
-  } else if (com.type > TOCK_SYSCALL_SUCCESS) {
-    return TOCK_FAIL;
-  } else {
-    return tock_error_to_rcode(com.data[0]);
-  }
+  return tock_command_return_novalue_to_returncode(com);
 }
 
 int ieee802154_get_address(unsigned short *addr) {
-  if (!addr) return TOCK_EINVAL;
+  if (!addr) return RETURNCODE_EINVAL;
+
   syscall_return_t com = command(RADIO_DRIVER, COMMAND_GET_ADDR, 0, 0);
-  if (com.type == TOCK_SYSCALL_SUCCESS_U32) {
-    // Driver adds 1 to make the value positive.
-    *addr = (unsigned short) (com.data[0] - 1);
-    return com.data[0];
-  } else if (com.type < TOCK_SYSCALL_SUCCESS) {
-    return tock_error_to_rcode(com.data[0]);
-  } else {
-    // incorrect success type
-    return TOCK_FAIL;
-  }
+  int ret = tock_command_return_u32_to_returncode(com, (uint32_t*) addr);
+
+  // Driver adds 1 to make the value positive.
+  *addr -= 1;
```

Comment:
outside of the scope of this PR, but would be nice to remove tricks like this necessitated by ReturnCode in the kernel

Reply:
I have no idea what this is doing.

Reply:
yeah, lets leave this for later. But I think it was because we were using SuccessWithValue, but sometimes the value we wanted to pass was 0, which would be indistinguishable from Success (or something like that)